crates/gitbutler-tauri/src/projects.rs

```diff
@@ -106,9 +106,20 @@ pub mod commands {
         project_id: ProjectId,
     ) -> Result<String, Error> {
         let project = projects.get(project_id)?;
-        let template_path = project.path.join(relative_path);
+        let template_path = project
+            .path
+            .join(relative_path)
+            .as_path()
+            .canonicalize()
+            .context("invalid template path")?;
 
-        Ok(read_file_from_workspace(template_path.as_path())?)
+        // Ensure that the merged path is a child of the project's root path
+        // in order to avoid path traversal issues, for example.
+        if template_path.starts_with(&project.path) {
+            Ok(read_file_from_workspace(template_path.as_path())?)
```

Comment:
If I could make one suggestion, make the `read_file_from_workspace` function take a project path as well as a relative path? Right now that function is misnamed, it'll read a file from anywhere while the name suggests it would only read it from a workspace.

Reply:
Yeah good call, I moved the checking logic to `read_file_from_workspace` and had it take the base path and relative file path :+1: